### PR TITLE
getAllReviews in T8 refactored to accept queries

### PR DESCRIPTION
**Pull request by BINQIU1220**

### CODE REVIEW

File: __tests__/reviews.test.js [hunk 1]
```diff
@@ -126,36 +126,141 @@ describe("PATCH /api/reviews/:review_id", () => {
...
   });
+  it("status:404, responds with 'Not Found' message when passed in a non-existing category", () => {
+    return request(app)
+      .get("/api/reviews?category=banana")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not Found");
+      });
+  });
+  it("status:404, responds with 'Not Found' message when passed in an existing category but with typo", () => {
```
Comment (DCrawley94) [inline __tests__/reviews.test.js:230]: I wouldn't worry too much about capitalisation/whitespace in your URL :)

File: models/reviews-model.js [hunk 2]
```diff
@@ -28,13 +28,54 @@ exports.updateVotesById = (reviewId, incVotes) => {
...
+    "category",
+    "title",
+    "created_at",
+    "votes",
+    "owner",
+    "review_id",
+    "comment_count",
+  ];
+
+  const validCategories = [
```
Comment (DCrawley94) [inline models/reviews-model.js:47]: It's perfectly valid to check the sort by and the order by. 

However checking the categories like this is very brittle!
What if more topics are added in the future? Also what if there are difference between the test and dev data?

Do we have to check before running the query? What happens if we don't?

**Review by DCrawley94 — CHANGES_REQUESTED** [review]
Nicely done Bin, a couple of things for you to look at here

**Review by dappernerddesigns — APPROVED** [review]
Nice work of util functions here. Merge when ready.

**Pull request merged**